source/sdk/lang/Exception.ooc

```diff
@@ -1,397 +1,324 @@
+			case EXCEPTION_FLT_INEXACT_RESULT => "(FLT_INEXACT_RESULT) the result of a floating point operation cannot be represented as a fraction"
+			case EXCEPTION_FLT_INVALID_OPERATION => "(FLT_INVALID_OPERATION) other floating point error"
+			case EXCEPTION_FLT_OVERFLOW => "(FLT_OVERFLOW) exponent of a floating-point operation greater than allowed by the type"
+			case EXCEPTION_FLT_STACK_CHECK => "(FLT_STACK_CHECK) stack overflow or underflow as a result of a floating point operation"
+			case EXCEPTION_FLT_UNDERFLOW => "(FLT_UNDERFLOW) exponent of a floating-point operation less than allowed by the type"
+			case EXCEPTION_ILLEGAL_INSTRUCTION => "(ILLEGAL_INSTRUCTION) tried to execute an invalid instruction"
+			case EXCEPTION_IN_PAGE_ERROR => "(IN_PAGE_ERROR) tried to access a page that was not present and that the system failed to load"
+			case EXCEPTION_INT_DIVIDE_BY_ZERO => "(INT_DIVIDE_BY_ZERO) tried to divide an integer value by zero"
+			case EXCEPTION_INT_OVERFLOW => "(INT_OVERFLOW) integer operation caused to carry out the most significant bit"
+			case EXCEPTION_INVALID_DISPOSITION => "(INVALID_DISPOSITION) exception handler returned an invalid disposition"
+			case EXCEPTION_NONCONTINUABLE_EXCEPTION => "(NONCONTINUABLE_EXCEPTION) tried to continue after a non-continuable exception"
+			case EXCEPTION_PRIV_INSTRUCTION => "(PRIV_INSTRUCTION) tried to execute an instruction not allowed in the current machine mode"
+			case EXCEPTION_SINGLE_STEP => "(SINGLE_STEP) trace trap or other mechanism signaled that one action was executed"
+			case EXCEPTION_STACK_OVERFLOW => "(STACK_OVERFLOW) the thread used up its stack"
+			case => "(?) unknown exception code %lu" format(code as ULong)
+		}
+		stderr write(message). write('\n')
+		handler := BacktraceHandler get()
+		context := exceptionInfo@ ContextRecord as Pointer
+		backtrace := handler backtraceWithContext(context)
+		if (backtrace)
+			stderr write(handler backtraceSymbols(backtrace))
+		else
+			stderr write("[no backtrace] use a debugger!")
+		EXCEPTION_EXECUTE_HANDLER
+	}
+	_controlHandler: func (ctrlType: DWORD) -> Bool {
+		message := match ctrlType {
+			case CTRL_C_EVENT => "(CTRL_C_EVENT) A CTRL+C signal was received"
+			case CTRL_BREAK_EVENT => "(CTRL_BREAK_EVENT) A CTRL+BREAK signal was received"
+			case CTRL_CLOSE_EVENT => "(CTRL_CLOSE_EVENT) Console was closed or task was ended"
+			case CTRL_LOGOFF_EVENT => "(CTRL_LOGOFF_EVENT) User logged off"
+			case CTRL_SHUTDOWN_EVENT => "(CTRL_SHUTDOWN_EVENT) System shutting down"
+			case => "(?) an unknown control signal %u was received" format(ctrlType as UInt)
+		}
+		stderr write(message). write('\n')
+		// that's all we can do - trying to print a backtrace
+		// will crash on Windows Vista and later. And we can't block
+		// the signal either as the process will be terminated regardless
+		// of what we return...
+		// pass it on to the next handler...
+		false
```

Comment:
I'm not sure if it works or if we need it at all, I will experiment with it tomorrow